rpcs3/Emu/RSX/GL/GLGSRender.cpp

```diff
@@ -836,41 +837,7 @@ bool GLGSRender::load_program()
 
 	if (fragment_constants_size)
 	{
-		auto mapping = m_uniform_ring_buffer.alloc_from_reserve(fragment_constants_size, m_uniform_buffer_offset_align);
-		fragment_constants_offset = mapping.second;
-
-		static const __m128i mask = _mm_set_epi8(
-			0xE, 0xF, 0xC, 0xD,
-			0xA, 0xB, 0x8, 0x9,
-			0x6, 0x7, 0x4, 0x5,
-			0x2, 0x3, 0x0, 0x1);
-
-		//The shader may be the same, but the value of the constants (and the shader location in memory) may have changed
-		//Point to the current shader location, not the cached version
-		auto ucode = (const rsx::fragment_program::ucode_instr *)prog.fragment_shader.ucode_ptr;
-
-		auto dst = (const rsx::fragment_program::ucode_instr *)mapping.first;
-
-		for (const auto& constant : info.fragment_shader.decompiled->constants)
-		{
-			const void *src = ucode + u32(constant.id / sizeof(*ucode));
-
-			const __m128i &vector = _mm_loadu_si128((const __m128i*)src);
-			const __m128i &shuffled_vector = _mm_shuffle_epi8(vector, mask);
-			_mm_stream_si128((__m128i*)dst, shuffled_vector);
-
-			if (0)
-			{
-				float x = ((float*)dst)[0];
-				float y = ((float*)dst)[1];
-				float z = ((float*)dst)[2];
-				float w = ((float*)dst)[3];
-
-				LOG_WARNING(RSX, "fc%u = {%g, %g, %g, %g}", constant.id, x, y, z, w);
-			}
-
-			++dst;
-		}
+		m_prog_buffer.fill_fragment_constans_buffer({ reinterpret_cast<float*>(buf), gsl::narrow<int>(fragment_constants_size) }, fragment_program);
```

Comment:
You are missing the mapping for fragment constants. See https://github.com/RPCS3/rpcs3/pull/2098/files#diff-472ec451195cefea00b28da26e2d4511L840. buf should be set to mapping.first otherwise you'll overwrite the transform constants.